Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/OnUpdate.hs

```diff
@@ -92,13 +93,33 @@ parseEvent _ (OnUpdate.RideAssigned taEvent) = do
         vehicleModel = vehicle.model
       }
 parseEvent _ (OnUpdate.RideStarted rsEvent) = do
+  tagsGroup <- fromMaybeM (InvalidRequest "agent tags is not present in RideStarted Event.") rsEvent.fulfillment.tags
+  let tripStartLat :: Maybe Double =
+        readMaybe . T.unpack
+          =<< getTag "ride_start_location" "started_from_lat" tagsGroup
+  let tripStartLon :: Maybe Double =
+        readMaybe . T.unpack
+          =<< getTag "ride_start_location" "started_from_lon" tagsGroup
+  let tripStartLocation = case (tripStartLat, tripStartLon) of
+        (Just lat, Just lon) -> Just (Maps.LatLong lat lon)
+        (_, _) -> Nothing
   return $
     DOnUpdate.RideStartedReq
       { bppBookingId = Id rsEvent.id,
-        bppRideId = Id rsEvent.fulfillment.id
+        bppRideId = Id rsEvent.fulfillment.id,
+        ..
       }
 parseEvent _ (OnUpdate.RideCompleted rcEvent) = do
   tagsGroup <- fromMaybeM (InvalidRequest "agent tags is not present in RideCompleted Event.") rcEvent.fulfillment.tags
+  let tripEndLat :: Maybe Double =
+        readMaybe . T.unpack
+          =<< getTag "ride_end_location" "ended_from_lat" tagsGroup
+  let tripEndLon :: Maybe Double =
+        readMaybe . T.unpack
+          =<< getTag "ride_end_location" "ended_from_lon" tagsGroup
+  let tripEndLocation = case (tripEndLat, tripEndLon) of
```

Comment:
can be simplified
`let tripEndLocation = Maps.LatLong <S> tripEndLat <*> tripEndLon`

Reply:
done

Reply:
done

Reply:
done